### PR TITLE
Chore/typescript docs

### PR DESCRIPTION
**Pull request by aschenkel**
It ts great that we added typescript to our own example since there were cases where we were using two parameters on a method that only supported one 😄

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -32,28 +47,53 @@ function useNavigationComponentDidAppear(handler: (event: ComponentDidAppearEven
   }, [handler, componentId])
 }
 
+/**
+Registers listener to be called each time component disappears from screen (detached from the view heirarchy) [more info](https://wix.github.io/react-native-navigation/api/events/#componentdiddisappear)
```
Comment (ignacioola) [inline src/index.ts:51]: Doc is not correctly formatted, missing `*`

File: src/index.ts [hunk 2]
```diff
@@ -32,28 +47,53 @@ function useNavigationComponentDidAppear(handler: (event: ComponentDidAppearEven
...
+      }
+    )
 
     return () => subscription.remove()
   }, [handler, componentId])
 }
 
-function useNavigationCommand(handler: (name: string, params: any) => void, commandName?: string) {
+/**
+Registers listener to be called when a Navigation command (i.e push, pop, showModal etc) is invoked. [more info](https://wix.github.io/react-native-navigation/api/events/#registercommandlistener)
```
Comment (ignacioola) [inline src/index.ts:82]: Here too

File: src/index.ts [hunk 3]
```diff
@@ -67,9 +107,22 @@ function useNavigationCommand(handler: (name: string, params: any) => void, comm
   }, [handler, commandName])
 }
 
-function useNavigationCommandComplete(handler: (event: CommandCompletedEvent) => void, commandName?: string) {
+/**
+Registers listener to be called when a command finishes executing in native. If the command contains animations, for example pushed screen animation,) the listener is invoked after the animation ends. [more info]https://wix.github.io/react-native-navigation/api/events/#registercommandcompletedlistener)
```
Comment (ignacioola) [inline src/index.ts:111]: And here

File: src/index.ts [hunk 4]
```diff
@@ -83,28 +136,53 @@ function useNavigationCommandComplete(handler: (event: CommandCompletedEvent) =>
   }, [handler, commandName])
 }
 
+/**
+Registers listener to be called only on iOS pageSheet modal when swipeToDismiss flag is set to true and modal swiped down to dismiss. [more info](https://wix.github.io/react-native-navigation/api/events/#registermodalattemptedtodismisslistenerios-13-only)
```
Comment (ignacioola) [inline src/index.ts:140]: And here

File: src/index.ts [hunk 5]
```diff
@@ -83,28 +136,53 @@ function useNavigationCommandComplete(handler: (event: CommandCompletedEvent) =>
...
+      }
+    )
 
     return () => subscription.remove()
   }, [handler, componentId])
 }
 
-function useNavigationModalDismiss(handler: (event: ModalDismissedEvent) => void, componentId?: string) {
+/**
+Registers listener to be called when modal is dismissed. [more info](https://wix.github.io/react-native-navigation/api/events/#registermodaldismissedlistener)
```
Comment (ignacioola) [inline src/index.ts:171]: And here

File: src/index.ts [hunk 6]
```diff
@@ -118,9 +196,22 @@ function useNavigationModalDismiss(handler: (event: ModalDismissedEvent) => void
   }, [handler, componentId])
 }
 
-function useNavigationScreenPop(handler: (event: ScreenPoppedEvent) => void, componentId?: string) {
+/**
+Registers listener to be called when screen is popped. [more info](https://wix.github.io/react-native-navigation/api/events/#registerscreenpoppedlistener)
```
Comment (ignacioola) [inline src/index.ts:200]: And here

File: src/index.ts [hunk 7]
```diff
@@ -134,54 +225,107 @@ function useNavigationScreenPop(handler: (event: ScreenPoppedEvent) => void, com
   }, [handler, componentId])
 }
 
-function useNavigationBottomTabSelect(handler: (event: BottomTabSelectedEvent) => void) {
+/**
+Registers listener to be called when a BottomTab is selected by the user. [more info](https://wix.github.io/react-native-navigation/api/events/#registerbottomtabselectedlistener)
```
Comment (ignacioola) [inline src/index.ts:229]: And here

File: src/index.ts [hunk 8]
```diff
@@ -134,54 +225,107 @@ function useNavigationScreenPop(handler: (event: ScreenPoppedEvent) => void, com
...
   useLayoutEffect(() => {
     const subscription = Navigation.events().registerBottomTabSelectedListener(handler)
 
     return () => subscription.remove()
   }, [handler])
 }
 
-function useNavigationBottomTabPress(handler: (event: BottomTabPressedEvent) => void) {
+/**
+Registers listener to be called when a BottomTab is pressed by the user. [more info](https://wix.github.io/react-native-navigation/api/events/#registerbottomtabpressedlistener)
```
Comment (ignacioola) [inline src/index.ts:245]: And here

File: src/index.ts [hunk 9]
```diff
@@ -134,54 +225,107 @@ function useNavigationScreenPop(handler: (event: ScreenPoppedEvent) => void, com
...
   useLayoutEffect(() => {
     const subscription = Navigation.events().registerBottomTabPressedListener(handler)
 
     return () => subscription.remove()
   }, [handler])
 }
 
-function useNavigationBottomTabLongPress(handler: (event: BottomTabLongPressedEvent) => void) {
+/**
+Registers listener to be called when a BottomTab is long pressed by the user. [more info](https://wix.github.io/react-native-navigation/api/events/#registerbottomtablongpressedlistener)
```
Comment (ignacioola) [inline src/index.ts:261]: And here

File: src/index.ts [hunk 10]
```diff
@@ -134,54 +225,107 @@ function useNavigationScreenPop(handler: (event: ScreenPoppedEvent) => void, com
...
+) {
   useLayoutEffect(() => {
     const subscription = Navigation.events().registerBottomTabLongPressedListener(handler)
 
     return () => subscription.remove()
   }, [handler])
 }
 
+/**
+Registers listener to be called when a TopBar button is pressed by the user. [more info](https://wix.github.io/react-native-navigation/api/events/#navigationbuttonpressed-event)
```
Comment (ignacioola) [inline src/index.ts:277]: And here

File: src/index.ts [hunk 11]
```diff
@@ -134,54 +225,107 @@ function useNavigationScreenPop(handler: (event: ScreenPoppedEvent) => void, com
...
+      }
+    )
 
     return () => subscription.remove()
   }, [handler, componentId, buttonId])
 }
 
-function useNavigationSearchBarUpdate(handler: (event: SearchBarUpdatedEvent) => void, componentId?: string) {
+/**
+Registers listener to be called when a SearchBar from NavigationBar gets updated. [more info](https://wix.github.io/react-native-navigation/api/events/#searchbarupdated-ios-11-only)
```
Comment (ignacioola) [inline src/index.ts:314]: And here 🙄

File: src/index.ts [hunk 12]
```diff
@@ -195,28 +339,53 @@ function useNavigationSearchBarUpdate(handler: (event: SearchBarUpdatedEvent) =>
   }, [handler, componentId])
 }
 
+/**
+Registers listener to be called when the cancel button on the SearchBar from NavigationBar gets pressed. [more info](https://wix.github.io/react-native-navigation/api/events/#searchbarcancelpressed-ios-11-only)
```
Comment (ignacioola) [inline src/index.ts:343]: And here

File: src/index.ts [hunk 13]
```diff
@@ -195,28 +339,53 @@ function useNavigationSearchBarUpdate(handler: (event: SearchBarUpdatedEvent) =>
...
+      }
+    )
 
     return () => subscription.remove()
   }, [handler, componentId])
 }
 
-function useNavigationPreviewComplete(handler: (event: PreviewCompletedEvent) => void, componentId?: string) {
+/**
+Registers listener to be called when preview peek is completed. [more info](https://wix.github.io/react-native-navigation/api/events/#previewcompleted-ios-114-only)
```
Comment (ignacioola) [inline src/index.ts:374]: And here

File: package.json [hunk 14]
```diff
@@ -57,6 +57,7 @@
     "react": ">=16.8.3",
     "react-native": ">=0.59.0",
     "react-native-navigation": ">=6.5.0",
+    "@types/react-native-navigation": "2.0.0",
```
Comment (jpgarcia) [inline package.json:60]: Do we really need to add this dependency? 

AFAIK **react-native-navigation** provide the types as part of the library so you don't need to add the types from and external package.

image

**Comment by aschenkel** [comment]
@ignacioola I had added that package because typescript was not working correctly. It seemed to be related to [this issue](https://github.com/microsoft/vscode/issues/71751#issuecomment-480126916).

As suggested on the comment, I downgraded typescript a bit and it is working ok now.

**Comment by ignacioola** [comment]
> @ignacioola I had added that package because typescript was not working correctly. It seemed to be related to [this issue](https://github.com/microsoft/vscode/issues/71751#issuecomment-480126916).
> 
> As suggested on the comment, I downgraded typescript a bit and it is working ok now.

@aschenkel I think it's OK. Let's wait for @jpgarcia since the comment you answered was from him.

**Pull request merged**